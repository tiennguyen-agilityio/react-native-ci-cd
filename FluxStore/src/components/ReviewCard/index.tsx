import {memo, useMemo} from 'react';
import {Image, StyleSheet} from 'react-native';

import {useThemeStore} from '@/stores';
import {fontSizes, fontWeights, metrics} from '@/themes';
import {getShortTimeAgo} from '@/utils';

import Flex from '../Flex';
import Text from '../Text';
import Rating from '../Rating';

interface ReviewCardProps {
  avatarUrl: string;
  name: string;
  rating: number;
  comment: string;
  createdAt: number;
}

const ReviewCard = ({avatarUrl, name, rating, comment, createdAt}: ReviewCardProps) => {
  const {
    theme: {text, fonts},
  } = useThemeStore();

  const styles = useMemo(
    () =>
      StyleSheet.create({
        image: {
          width: 36,
          height: 36,
          borderRadius: 18,
          overflow: 'hidden',
        },
        name: {
          fontSize: fontSizes.xxs,
          fontWeight: fontWeights.bold,
          fontFamily: fonts.primary.bold,
        },
        time: {
          fontSize: fontSizes.micro,
          fontWeight: fontWeights.medium,
          fontFamily: fonts.secondary.medium,
          color: text.septenary,
        },
        comment: {
          fontSize: fontSizes.micro,
          color: text.default,
        },
      }),
    [text, fonts],
  );

  const timeAgo = useMemo(() => getShortTimeAgo(createdAt), [createdAt]);

  return (
    <Flex gap={metrics.dimensions.lg}>
      <Flex direction="row" gap={12} align="center">
        <Image source={{uri: avatarUrl}} resizeMode="cover" style={styles.image} />
        <Flex gap={8}>
          <Text style={styles.name}>{name}</Text>
          <Rating value={rating} size={10} />
        </Flex>
        <Flex marginLeft="auto" marginRight={0}>
          <Text style={styles.time}>{timeAgo}</Text>
        </Flex>
      </Flex>
      <Text style={styles.comment}>{comment}</Text>
    </Flex>
  );
};

export default memo(ReviewCard);
