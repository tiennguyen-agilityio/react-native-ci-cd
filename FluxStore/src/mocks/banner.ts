export const BANNERS = [
  {
    id: '01',
    title: 'Clothing Collection 2025',
    imageUrl: 'https://i.ibb.co/wZv2s1T6/Frame-33149.png',
  },
  {
    id: '02',
    title: 'Autumn Collection 2025',
    imageUrl: 'https://i.ibb.co/hRBFHvHk/Mask-Group-1.png',
  },
  {
    id: '03',
    title: 'Accessories Collection 2025',
    imageUrl: 'https://i.ibb.co/20WCrpXW/Frame-33123.png',
  },
];
