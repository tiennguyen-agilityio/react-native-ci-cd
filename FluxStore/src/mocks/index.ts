export * from './product';
export * from './banner';
export * from './review';
export * from './cart';
