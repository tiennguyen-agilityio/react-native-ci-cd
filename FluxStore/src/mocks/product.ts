import {Product} from '@/interfaces';

export const COLORS = ['black', 'red', 'blue'];
export const SIZES = ['S', 'M', 'L'];

export const PRODUCTS: Product[] = [
  {
    id: '01',
    name: 'Linen Dress',
    price: 90.02,
    discount: 10,
    rating: 5,
    reviewCount: 20,
    image: 'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['s', 'M'],
    colors: ['black', 'red', 'blue'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
  {
    id: '02',
    name: 'Filted Waist Dress',
    price: 82.0,
    discount: 0,
    rating: 3,
    reviewCount: 110,
    image: 'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['XL'],
    colors: ['pink', 'black'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
];
