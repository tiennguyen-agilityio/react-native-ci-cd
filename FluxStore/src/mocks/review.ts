export const REVIEWS = [
  {
    id: '01',
    avatarUrl: 'https://i.ibb.co/h1n0FRHn/Ellipse-76.png',
    name: 'Jennifer Rose',
    rating: 5,
    comment:
      'I love it.  Awesome customer service!! Helped me out with adding an additional item to my order. Thanks again!',
    createdAt: 1747710042553,
  },
  {
    id: '02',
    avatarUrl: 'https://i.ibb.co/tT8QbLVp/Ellipse-77.png',
    name: 'Kelly Rihana',
    rating: 5,
    comment: "I'm very happy with order, It was delivered on and good quality. Recommended!",
    createdAt: 1747710042570,
  },
];
