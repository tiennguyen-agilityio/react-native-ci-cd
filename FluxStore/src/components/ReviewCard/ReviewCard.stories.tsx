import {Meta, StoryObj} from '@storybook/react';

import {REVIEWS} from '@/mocks';

import ReviewCard from '.';

const meta: Meta<typeof ReviewCard> = {
  title: 'Components/ReviewCard',
  component: ReviewCard,
};

export default meta;

type Story = StoryObj<typeof ReviewCard>;

export const Default: Story = {
  args: {
    avatarUrl: REVIEWS[0].avatarUrl,
    name: REVIEWS[0].name,
    rating: REVIEWS[0].rating,
    comment: REVIEWS[0].comment,
    createdAt: REVIEWS[0].createdAt,
  },
};
