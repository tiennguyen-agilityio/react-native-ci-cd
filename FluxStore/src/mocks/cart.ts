import {PRODUCTS} from './product';

export const CARTS = [
  {
    id: '01',
    product: {
      id: PRODUCTS[0].id,
      image: PRODUCTS[0].image,
      name: PRODUCTS[0].name,
      price: PRODUCTS[0].price,
    },
    sizes: [PRODUCTS[0]?.sizes?.[0], PRODUCTS[0]?.sizes?.[1]].join(', '),
    colors: [PRODUCTS[0]?.colors?.[0]].join(', '),
    quantity: 2,
    isChecked: true,
  },
  {
    id: '02',
    product: {
      id: PRODUCTS[1].id,
      image: PRODUCTS[1].image,
      name: PRODUCTS[1].name,
      price: PRODUCTS[1].price,
    },
    sizes: [PRODUCTS[1]?.sizes?.[0]].join(', '),
    colors: [PRODUCTS[1]?.colors?.[0], PRODUCTS[1]?.colors?.[1]].join(', '),
    quantity: 1,
    isChecked: false,
  },
];
