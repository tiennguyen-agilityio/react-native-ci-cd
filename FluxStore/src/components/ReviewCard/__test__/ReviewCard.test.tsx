import {render} from '@testing-library/react-native';

import ReviewCard from '..';
import {REVIEWS} from '@/mocks';

describe('ReviewCard', () => {
  const props = {
    avatarUrl: REVIEWS[0].avatarUrl,
    name: REVIEWS[0].name,
    rating: REVIEWS[0].rating,
    comment: REVIEWS[0].comment,
    createdAt: new Date().getTime(),
  };

  it('should render ReviewCard component', () => {
    const {toJSON} = render(<ReviewCard {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
