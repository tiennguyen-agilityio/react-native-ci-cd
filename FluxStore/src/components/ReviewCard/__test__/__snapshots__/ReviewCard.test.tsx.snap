// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard should render ReviewCard component 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      undefined,
      {
        "gap": 20,
      },
    ]
  }
  testID="flex"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "center",
        },
        {
          "flexWrap": "nowrap",
        },
        undefined,
        {
          "gap": 12,
        },
      ]
    }
    testID="flex"
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "https://i.ibb.co/h1n0FRHn/Ellipse-76.png",
        }
      }
      style={
        {
          "borderRadius": 18,
          "height": 36,
          "overflow": "hidden",
          "width": 36,
        }
      }
    />
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "flex-start",
          },
          {
            "alignItems": "stretch",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "gap": 8,
          },
        ]
      }
      testID="flex"
    >
      <Text
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#737680",
              "fontFamily": "ProductSansRegular",
              "fontSize": 14,
            },
            {},
            {
              "fontFamily": "ProductSansBold",
              "fontSize": 13,
              "fontWeight": 700,
            },
          ]
        }
      >
        Jennifer Rose
      </Text>
      <View
        style={
          {
            "columnGap": 5,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="star-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={10}
            bbWidth={10}
            fill="none"
            focusable={false}
            height={10}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 10,
                  "width": 10,
                },
              ]
            }
            vbHeight={8}
            vbWidth={9}
            width={10}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                clipRule={0}
                d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
                fill={
                  {
                    "payload": 4283468411,
                    "type": 0,
                  }
                }
                fillRule={0}
                propList={
                  [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="star-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={10}
            bbWidth={10}
            fill="none"
            focusable={false}
            height={10}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 10,
                  "width": 10,
                },
              ]
            }
            vbHeight={8}
            vbWidth={9}
            width={10}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                clipRule={0}
                d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
                fill={
                  {
                    "payload": 4283468411,
                    "type": 0,
                  }
                }
                fillRule={0}
                propList={
                  [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="star-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={10}
            bbWidth={10}
            fill="none"
            focusable={false}
            height={10}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 10,
                  "width": 10,
                },
              ]
            }
            vbHeight={8}
            vbWidth={9}
            width={10}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                clipRule={0}
                d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
                fill={
                  {
                    "payload": 4283468411,
                    "type": 0,
                  }
                }
                fillRule={0}
                propList={
                  [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="star-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={10}
            bbWidth={10}
            fill="none"
            focusable={false}
            height={10}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 10,
                  "width": 10,
                },
              ]
            }
            vbHeight={8}
            vbWidth={9}
            width={10}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                clipRule={0}
                d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
                fill={
                  {
                    "payload": 4283468411,
                    "type": 0,
                  }
                }
                fillRule={0}
                propList={
                  [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="star-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={10}
            bbWidth={10}
            fill="none"
            focusable={false}
            height={10}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 10,
                  "width": 10,
                },
              ]
            }
            vbHeight={8}
            vbWidth={9}
            width={10}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                clipRule={0}
                d="M6.097 2.624a.328.328 0 01-.313-.22L5.058.22a.332.332 0 00-.625 0l-.73 2.18a.328.328 0 01-.313.22H1.025a.32.32 0 00-.311.216.313.313 0 00.117.357l1.92 1.359a.31.31 0 01.118.352L2.137 7.1a.315.315 0 00.166.383.331.331 0 00.34-.03l1.907-1.35a.339.339 0 01.389 0l1.906 1.349a.331.331 0 00.34.03.326.326 0 00.178-.232.315.315 0 00-.011-.15L6.62 4.906a.311.311 0 01.118-.352L8.658 3.2a.313.313 0 00-.002-.516.321.321 0 00-.191-.057l-2.368-.003z"
                fill={
                  {
                    "payload": 4283468411,
                    "type": 0,
                  }
                }
                fillRule={0}
                propList={
                  [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "flex-start",
          },
          {
            "alignItems": "stretch",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "marginLeft": "auto",
            "marginRight": 0,
          },
        ]
      }
      testID="flex"
    >
      <Text
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#737680",
              "fontFamily": "ProductSansRegular",
              "fontSize": 14,
            },
            {},
            {
              "color": "#bebfc4",
              "fontFamily": "NunitoSemiBold",
              "fontSize": 11,
              "fontWeight": 500,
            },
          ]
        }
      >
        0m ago
      </Text>
    </View>
  </View>
  <Text
    style={
      [
        {
          "letterSpacing": 0,
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#737680",
          "fontFamily": "ProductSansRegular",
          "fontSize": 14,
        },
        {},
        {
          "color": "#000",
          "fontSize": 11,
        },
      ]
    }
  >
    I love it.  Awesome customer service!! Helped me out with adding an additional item to my order. Thanks again!
  </Text>
</View>
`;
